### PR TITLE
fix: remove unnecessary typing-extensions for py3.8

### PR DESCRIPTION
**Pull request by nadobando**
**Issue #280**

## Description of changes:
1. Added if block to know if typing_extensions should be imported based on the python version.
if python >= 3.8 typing extensions will not be imported and the native implementation will be used
else will try to import typing_extensions and catch the ImportError

2. Updated modules which imported this dependency.

3. Updated pyproject.toml to only install typing-extensions when python version is <3.8
**Checklist**

 

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [ ] Update tests
* [ ] Update docs
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

## Breaking change checklist

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws_lambda_powertools/utilities/parser/types.py [hunk 1]
```diff
@@ -1,6 +1,16 @@
 """Generics and other shared types used across parser"""
+import sys
```
Comment (cakepietoast) [inline aws_lambda_powertools/utilities/parser/types.py:2]:
```suggestion
```

File: aws_lambda_powertools/utilities/parser/types.py [hunk 2]
```diff
@@ -1,6 +1,16 @@
...
 from pydantic import BaseModel
 
+# Workaround for not importing typing_extensions on python ^3.8
+if sys.version_info[0] > 3 or (sys.version_info[0] == 3 and sys.version_info[1] >= 8):
+    from typing import Literal  # noqa: F401
+else:
+    try:
+        from typing_extensions import Literal  # noqa: F401
+    except ImportError:
+        raise Exception("please install typing-extensions or upgrade to Python >= 3.8")
```
Comment (cakepietoast) [inline aws_lambda_powertools/utilities/parser/types.py:14]: We can simplify this and just fall back to typing_extensions if the import of `typing.Literal` fails. No need to check the python version or re-raise the ImportError.


```suggestion
# We only need typing_extensions for python versions <3.8
try:
    from typing import Literal  # noqa: F401
except ImportError:
    from typing_extensions import Literal  # noqa: F401
```

**Review by cakepietoast — CHANGES_REQUESTED** [review]
Nice change, great to keep the package size down where we can! Added a small suggestion to change - otherwise LGTM!

File: aws_lambda_powertools/utilities/parser/types.py (hunk 2 above)
Comment (nadobando) [reply aws_lambda_powertools/utilities/parser/types.py:14]: Sure, I will do it 👍

**Review by cakepietoast — APPROVED** [review]

**Comment by heitorlessa** [comment]
Thanks a lot for this contribution @nadobando !!! And thanks @cakepietoast for the review. Merging this and publishing a new release

**Pull request merged**